### PR TITLE
fix: Improve and refactor `BaseSkel.fromClient()` to handle empty/unset data

### PR DESCRIPTION
**Pull request by phorward**
Some necessary improvements to `RelSkel.fromClient()` and `BaseSkeleton.fromClient()` for integration with the new vi-admin.

- RelSkel.fromClient() is valid when no data is provided and all fields are not required
- errors are now being iterated only once
- better "incomplete" detection
- comments (huh, woooow!)
- removed `allowEmptyRequired`-parameter from all functions as it is not necessary anymore
- tested against normal bones, RelationalBones with using skel, and RecordBones with several bone configurations

### CODE REVIEW

**Comment by phorward** [comment]
Hello, there are some edge cases where the validation might not be as expected, e.g. when there is a `RecordBone(multiple=False)`, which is not required, but has a required field. In this case, the RecordBone will be flagged invalid when any value for the RecordBone was provided, but not a required one.

By the current design, this behavior is correct: The RecordBone itself is not required, therefore its content is not required. When its content is provided, but invalid, is is correctly rejected.

I'm sure this won't break the whole system now, as several tests worked correctly and as expected. The handling is now better, as vi-admin does not send fields which aren't there.

I would like to bring this into viur-core 3.5.11 asap, as it is required for a project.

**Comment by phorward** [comment]
The latest commit entirely removes `RelSkel.fromClient()` as it becomes equal to `BaseSkel.fromClient()`, and the custom stuff is done in subclassed `Skeleton.fromClient()`. Tests worked.

**Review by sveneberth — CHANGES_REQUESTED** [review]
I tested this with `RelationBones` and couldn't find any error which is caused by this change.

File: src/viur/core/skeleton.py [hunk 1]
```diff
@@ -707,9 +720,17 @@ def fromClient(cls, skelValues: SkeletonInstance, data: Dict[str, Union[List[str
...
+
         # Load data into this skeleton
-        complete = super().fromClient(skelValues, data)
+        complete = bool(data) and super().fromClient(skelValues, data)
+
+        if (
+            not data  # in case data is empty
+            or (len(data) == 1 and "key" in data)
+            # FIXME: Use case? Otherwise, use conf.viur.bone.boolean.str2true!
+            or (str(data.get("nomissing")) == "1")
```
[the remark this answers is not in the document]
Comment (phorward) [reply src/viur/core/skeleton.py:731]: Yes, thanks. Indeed, IMHO such special parameters should use a (configurable) "viur-"-prefix or similar, or "nomissing" must be reserved name.

**Review by sveneberth — APPROVED** [review]

**Review by ArneGudermann — APPROVED** [review]

**Pull request merged**